### PR TITLE
Security app initial works for brute force protection and strong password enforcement

### PR DESCRIPTION
**Pull request by karakayasemi**
To detect failed login attempts loginFailed hook is used.

### CODE REVIEW

File: appinfo/info.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    <author>Semih Serhat Karakaya</author>
+    <version>0.0.1</version>
+    <namespace>Security</namespace>
+    <dependencies>
+        <owncloud min-version="9.0" max-version="10.2" />
+    </dependencies>
+    <types>
+        <prelogin/>
+    </types>
+    <category>tool</category>
```
Comment (DeepDiver1975) [inline appinfo/info.xml]: security

File: appinfo/info.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0"?>
+<info>
+    <id>security</id>
+    <name>Security</name>
+    <description>OwnCloud security app. It harden ownCloud security with several new features like brute force protection, 404 detection, strong password enforcement.</description>
```
Comment (DeepDiver1975) [inline appinfo/info.xml]: O -> o

File: appinfo/info.xml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0"?>
+<info>
+    <id>security</id>
+    <name>Security</name>
```
Comment (DeepDiver1975) [inline appinfo/info.xml]: @pmaier1 we need your input on summary and description

File: lib/Hooks.php [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        $ip = \OC::$server->getRequest()->getRemoteAddress();
+		$this->throttle->addFailedLoginAttempt($uid, $ip);
+		$this->throttle->putDelay($uid, $ip);
+    }
+
+    /**
+     * @param IUser $user
+     */
+    private function postLoginCallback($user) {
+		$ip = \OC::$server->getRequest()->getRemoteAddress();
```
Comment (DeepDiver1975) [inline lib/Hooks.php]: indent

File: lib/Hooks.php [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+            $this->postLoginCallback($user);
+        });
+    }
+
+    /**
+     * @param string $uid
+     * @param Throttle $throttle
+     */
+    private function failedLoginCallback($uid) {
+        $ip = \OC::$server->getRequest()->getRemoteAddress();
```
Comment (DeepDiver1975) [inline lib/Hooks.php]: you should inject the request object in the ctor

File: appinfo/info.xml (hunk 3 above)
Comment (pmaier1) [inline appinfo/info.xml]: Basically I think for community apps it's the author's right/responsibility to provide these information. If not applicable for some reason I'm fine proposing summary and description but I would need to try the app first, of course. Just tell me!

File: appinfo/info.xml [hunk 6]
```diff
@@ -2,7 +2,7 @@
 <info>
     <id>security</id>
     <name>Security</name>
-    <description>OwnCloud security app. It harden ownCloud security with several new features like brute force protection, 404 detection, strong password enforcement.</description>
+    <description>ownCloud security app. It harden ownCloud security with several new features like brute force protection, 404 detection, strong password enforcement.</description>
```
Comment (tomneedham) [inline appinfo/info.xml]: ownCloud Security app. It hardens ownCloud security with several new features including brute force protection, 404 detection and strong password enforcement.

File: .travis.yml [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  # execute unit tests
+  - cd apps/security/tests
+  - ../../../lib/composer/phpunit/phpunit/phpunit --configuration phpunit.xml
+
+matrix:
+  include:
+    - php: 5.6
+      env: DB=mysql
+    - php: 7.0
+      env: DB=mysql
```
Comment (DeepDiver1975) [inline .travis.yml]: please also test postgres

File: appinfo/info.xml [hunk 8]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0"?>
+<info>
+    <id>security</id>
+    <name>Security</name>
+    <description>ownCloud Security app. It hardens ownCloud security with several new features including brute force protection, 404 detection and strong password enforcement.</description>
```
Comment (DeepDiver1975) [inline appinfo/info.xml]: @karakayasemi please add a summary tag as well - THX

**Comment by karakayasemi** [comment]
Summary tag and PostgreSQL tests are added.

**Comment by karakayasemi** [comment]
@DeepDiver1975 please review. Thanks.

**Pull request merged**